src/main/java/com/example/demo/Controllers/FilesController.java

```diff
@@ -0,0 +1,137 @@
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api/test/files")
+public class FilesController {
+
+  private final FileService fileService;
+  private final UserRepository userRepository;
+
+  @Autowired
+  public FilesController(FileService fileService, UserRepository userRepository) {
+    this.fileService = fileService;
+    this.userRepository = userRepository;
+  }
+
+
+//  @PreAuthorize("hasRole('USER')")
+//  @GetMapping("")
+//    public String testing(){
+//      return "hello";
+//  }
+
+  @PreAuthorize("hasRole('USER')")
+  @PostMapping("/add/{userId}")
+  public ResponseEntity<String> upload(@PathVariable UUID userId, @RequestParam("file") MultipartFile file) {
+    try {
+
+      fileService.save(file, userId);
+      return ResponseEntity.status(HttpStatus.OK).body(String.format("File uploaded successfully: %s", file.getOriginalFilename()));
+    } catch (Exception e) {
+      return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(String.format("Could not upload the file: %s", file.getOriginalFilename()));
+    }
+
+  }
+
+  @PreAuthorize("hasRole('USER')")
+  @GetMapping("")
+  public List<FileResponse> list() {
+    return fileService.getAllFiles().stream().map(this::mapToFileResponse).collect(Collectors.toList());
+  }
+
+  //
+  @PreAuthorize("hasRole('USER')")
+  @GetMapping("/{userId}")
+  public List<Files> getFilesOfUser(@PathVariable UUID userId) {
+    return fileService.getFilesByUserId(userId);
+//      .stream().map(this::mapToFileResponse).collect(Collectors.toList());
+//      System.out.println(fileResponses);
+////       Optional<User> user=userRepository.findById(userId);
+////       if(user.isPresent()){
+////        for(FileResponse f:fileResponses){
+////          f.setUid(f.getUser().getId());
+////        }
+////        System.out.println(fileResponses);
+//         return ResponseEntity.status(HttpStatus.OK).body(fileResponses);
+//       }
+//       else return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("user not found with this user id");
+
+  }
+
+
+  private FileResponse mapToFileResponse(Files file) {
+    String downloadUrl = ServletUriComponentsBuilder.fromCurrentContextPath().path("/files/").path(file.getId()).toUriString();
+    FileResponse fileResponse = new FileResponse();
+    fileResponse.setId(file.getId());
+    fileResponse.setName(file.getName());
+    fileResponse.setContentType(file.getContentType());
+    fileResponse.setSize(file.getSize());
+    fileResponse.setCreatedAt(file.getCreatedAt());
+    fileResponse.setUpdatedAt(file.getUpdatedAt());
+    fileResponse.setUid(file.getUser().getId());
+    fileResponse.setUserName(file.getUser().getUsername());
+    fileResponse.setUrl(downloadUrl);
+
+    return fileResponse;
+  }
+
+  @PreAuthorize("hasRole('USER')")
+  @DeleteMapping("delete/{id}")
+  public ResponseEntity<?> deleteFile(@PathVariable String id) {
```

Comment:
ResponseEntity<?> - ? is used specifically in cases when we don't know the return type. Here we can configure the return type according to our needs, therefore we can specify the type here.